src/utils/geometry.rs

```diff
@@ -194,6 +227,31 @@ macro_rules! floating_point_coordinate_impl {
             fn abs(self) -> Self {
                 self.abs()
             }
+
+            #[inline]
+            fn saturating_add(self, other: Self) -> Self {
+                match self + other {
+                    x if x == Self::INFINITY => Self::MAX,
+                    x if x == Self::NEG_INFINITY => Self::MIN,
+                    x => x,
+                }
+            }
+            #[inline]
+            fn saturating_sub(self, other: Self) -> Self {
+                match self - other {
+                    x if x == Self::INFINITY => Self::MAX,
+                    x if x == Self::NEG_INFINITY => Self::MIN,
+                    x => x,
+                }
+            }
+            #[inline]
+            fn saturating_mul(self, other: Self) -> Self {
+                match self * other {
+                    x if x == Self::INFINITY => Self::MAX,
+                    x if x == Self::NEG_INFINITY => Self::MIN,
+                    x => x,
+                }
+            }
```

Comment:
I'm not quite sure about this one, feels wrong to clamp infinite values on float. :thinking:

Reply:
Yeah, I wasn't sure either, but Infinity is not a very useful value inside a compositor for coordinates. I can easily drop that though and just do the normal operations.

Reply:
Well, `1e306` is not a useful value either, but at least infinity (and its tendency to produce NaN) can make potential bugs more obvious I'd say

Reply:
Granted. :+1: